tests/e2e/e2e_lsm_test.go

```diff
@@ -0,0 +1,204 @@
+			s.Require().NoError(err)
+
+			return amt.Equal(sdk.NewDecFromInt(delegationAmount.Sub(tokenizeAmount)))
+		},
+		20*time.Second,
+		5*time.Second,
+	)
+
+	// Validate balance increased
+	recordID := int(1)
+	shareDenom := fmt.Sprintf("%s/%s", strings.ToLower(validatorAddressA), strconv.Itoa(recordID))
+	s.Require().Eventually(
+		func() bool {
+			res, err := getSpecificBalance(chainEndpoint, delegatorAddress, shareDenom)
+			s.Require().NoError(err)
+			return res.Amount.Equal(tokenizeAmount)
+		},
+		20*time.Second,
+		5*time.Second,
+	)
+
+	// Bank send LSM token
+	sendAmount := sdk.NewCoin(shareDenom, tokenizeAmount)
+	s.execBankSend(s.chainA, 0, delegatorAddress, validatorAAddr.String(), sendAmount.String(), standardFees.String(), false)
+
+	// Validate tokens are sent properly
+	s.Require().Eventually(
+		func() bool {
+			afterSenderShareDenomBalance, err := getSpecificBalance(chainEndpoint, delegatorAddress, shareDenom)
+			s.Require().NoError(err)
+
+			afterRecipientShareDenomBalance, err := getSpecificBalance(chainEndpoint, validatorAAddr.String(), shareDenom)
+			s.Require().NoError(err)
+
+			decremented := afterSenderShareDenomBalance.IsNil() || afterSenderShareDenomBalance.IsZero()
+			incremented := afterRecipientShareDenomBalance.IsEqual(sendAmount)
+
+			return decremented && incremented
+		},
+		time.Minute,
+		5*time.Second,
+	)
+
+	// transfer reward ownership
+	s.executeTransferTokenizeShareRecord(s.chainA, 0, strconv.Itoa(recordID), delegatorAddress, validatorAAddr.String(), gaiaHomePath, standardFees.String())
+
+	// Validate ownership transferred correctly
+	s.Require().Eventually(
+		func() bool {
+			record, err := queryTokenizeShareRecordByID(chainEndpoint, recordID)
+			s.Require().NoError(err)
+			return record.Owner == validatorAAddr.String()
+		},
+		time.Minute,
+		5*time.Second,
+	)
+
+	// IBC transfer LSM token
+	ibcTransferAmount := sdk.NewCoin(shareDenom, sdk.NewInt(100000000))
+	s.sendIBC(s.chainA, 0, validatorAAddr.String(), validatorBAddr.String(), ibcTransferAmount.String(), standardFees.String(), "memo")
+
+	s.Require().Eventually(
+		func() bool {
+			afterSenderShareBalance, err := getSpecificBalance(chainEndpoint, validatorAAddr.String(), shareDenom)
+			s.Require().NoError(err)
+
+			decremented := afterSenderShareBalance.Add(ibcTransferAmount).IsEqual(sendAmount)
+			return decremented
+		},
+		1*time.Minute,
+		5*time.Second,
+	)
+
+	// Redeem tokens for shares
+	s.executeRedeemShares(s.chainA, 0, sendAmount.Sub(ibcTransferAmount).String(), validatorAAddr.String(), gaiaHomePath, fees.String())
+
+	// check redeem success
+	s.Require().Eventually(
+		func() bool {
+			delegationRes, err := queryDelegation(chainEndpoint, validatorAddressA, validatorAAddr.String())
+			delegation := delegationRes.GetDelegationResponse().GetDelegation()
+			s.Require().NoError(err)
+
+			balanceRes, err := getSpecificBalance(chainEndpoint, delegatorAddress, shareDenom)
```

Comment:
should this be a check on validatorAAddress instead of delegatorAddress?

Reply:
Will update